common/src/chain/order/mod.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization::{Decode, Encode};
+
+use super::{output_value::OutputValue, Destination};
+
+mod order_id;
+pub use order_id::{make_order_id, OrderId};
+
+mod rpc;
+pub use rpc::RpcOrderInfo;
+
+/// Order data provides unified data structure to represent an order.
+/// There are no buy or sell types of orders per se but rather exchanges.
+/// The fields represent currencies and amounts to be exchanged and the trading pair can be deducted from it.
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, serde::Serialize, serde::Deserialize)]
+pub struct OrderData {
+    /// The key that can authorize conclusion of an order
```

Comment:
I think it would be good to explain here what concluding an order will result into, for anyone reading this.